salt/cloud/clouds/digital_ocean.py

```diff
@@ -694,17 +738,25 @@ def destroy(name, call=None):
     data = show_instance(name, call='action')
     node = query(method='droplets', droplet_id=data['id'], http_method='delete')
 
-    delete_record = config.get_cloud_config_value(
-        'delete_dns_record', get_configured_provider(), __opts__, search_global=False, default=None,
-    )
+    ## This is all terribly optomistic:
+    # vm_ = get_vm_config(name=name)
+    # delete_dns_record = config.get_cloud_config_value(
+    #     'delete_dns_record', vm_, __opts__, search_global=False, default=None,
+    # )
+    delete_dns_record = True
 
-    if delete_record and not isinstance(delete_record, bool):
+    if delete_dns_record and not isinstance(delete_dns_record, bool):
```

Comment:
If you're setting `delete_dns_record = True` above, then this check isn't even necessary.